### DISCUSSION TITLE
Large data structures in the code area

### DISCUSSION THREAD

**Discussion by IanWitham** [jkotlinski/durexforth#502] [Q&A] [answered]
I'm implementing a wave function collapse algorithm, for which I'd like to utilise the "code" area of memory for some large data structures, (a heap and a lookup table, each 2kb in size). The problem I have is that I'm coding in v and every time I compile the file, it tries to reallocate that space, which quickly leads to problems. Is there a way to only allocate that space once, and not on successive compiles? (I'm sure there is, but it's beyond my ability at the moment). I'm currently using the "high res graphics" area for these data structures, but now I'd like to try to implement backtracking I'm going to need more memory!
Preliminary results are attached :) 
![vice-screen-2023010409405847](https://user-images.githubusercontent.com/1018548/210440188-b833962a-bf23-4124-bf75-6696de8708be.png)

**Comment by Whammo** [comment, marked as answer]
You can use separate files, data and code. then only compile what you require.

**Reply by IanWitham** [reply]
Thanks, this works 👍

**Reply by Whammo** [reply]
That's the great thing about forth. However you do it, it just works.

**Comment by jkotlinski** [comment]
I think there are many ways to achieve what you want. Some ways on top of my head:

- Write the code that allocates the large buffers in a separate file. Then REQUIRE that file, to make sure it executes only once.
- Allocate the large buffers, then save a new forth using SAVE-FORTH. Use that forth from now on.
- Place your large buffers at a fixed address, for example $c000.

Cheers!
